dot-gen/src/main/java/com/wipro/ats/bdre/ldg/GetDotForTable.java

```diff
@@ -0,0 +1,126 @@
+    public String dotGeneratorWithCol(String[] args) {
+        if(args.length !=2) {
+            LOGGER.info("Two arguments required: please Give both Column name and Table name");
+            System.exit(-1);
+        }
+        GetDotForTable getDotForTable = new GetDotForTable();
+        GetNodeIdForLineageRelation getNodeIdForLineageRelation = new GetNodeIdForLineageRelation();
+
+        List<LineageRelation> lineageRelationList;
+        List<LineageRelation> lineageRelationListForTargetNodes;
+
+        //get the nodes which have the same DISPLAY_NAME as that of the string(COLUMN NAME) provided
+
+        GetLineageNodeByColName getLineageNodeByColName;
+        LineageMain lineageMain = new LineageMain();
+        //src table name is provided by user
+        String srcTableName = args[1];
+        LOGGER.info("Generating DOT for the given Column Name: " + args[0]);
+        String targetTableName = null;
+        StringBuffer relationDot = new StringBuffer("");
+        getLineageNodeByColName = new GetLineageNodeByColName();
+        //for each relation a new node is created in LN table, so get all those
+        List<LineageNode> srcTableNodeList = getLineageNodeByColName.getTableDotFromTableName(srcTableName);
+        List<LineageNode> lineageNodeList = new ArrayList<LineageNode>();
+        int counter =0;
+        //now get all those col nodes corresponding to the tables created for each relation
+        for(LineageNode lnodes:srcTableNodeList) {
+            LOGGER.info("----------------------------------src table" + lnodes.getDisplayName());
+             lineageNodeList.add(getDotForTable.getLineageNodeNodeId(args[0], lnodes));
+            LOGGER.info("----------------------------------col node" + lineageNodeList.get(counter++).getDisplayName());
+        }
+        StringBuffer targetTableNode = new StringBuffer("");
+
+        //compare the list of Nodes got from LN table with LR table to get the required node
+        for (LineageNode lineageNode : lineageNodeList) {
+            //this only checks the case where the node is src
+            LOGGER.info("------------------------inside");
+            lineageRelationList = getNodeIdForLineageRelation.execute(lineageNode.getNodeId());
+            for(LineageRelation lineageRelation:lineageRelationList) {
+                LOGGER.info("------------------------inside src check");
+                relationDot.append("\n" + lineageRelation.getDotString() + "\n");                                            //search for target table name in the DOT string
+                LOGGER.info("------------------" + relationDot);
+                targetTableNode.append("\n" + getLineageNodeByColName.getTableDotFromNodeId(lineageRelation.getLineageNodeByTargetNodeId(), srcTableNodeList.get(0)) + "\n");
+                LOGGER.info("-------------------------" + targetTableNode);
+                if(targetTableNode.equals("same-nodes")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "same-nodes" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1132)